src/main/java/org/kaazing/netx/ws/internal/WebSocketInputStateMachine.java

```diff
@@ -56,143 +51,73 @@
-        }
-    }
-
-    public void processPong(final WsURLConnectionImpl connection,
-                            final Pong pongFrame,
-                            final WebSocketFrameConsumer terminalConsumer) throws IOException {
-        WebSocketExtensionSpi sentinel = new WebSocketExtensionSpi() {
-            {
-                onPongFrameReceived = terminalConsumer;
-            }
-        };
-
-        WebSocketContext context = connection.getContext(sentinel, false);
-        WebSocketState state = connection.getInputState();
+        context.reset();
 
         switch (state) {
         case OPEN:
-            transition(connection, WebSocketTransition.RECEIVED_PONG_FRAME);
-            context.onPongReceived(pongFrame);
-            break;
-        default:
-            transition(connection, WebSocketTransition.ERROR);
-            context.onError(format("Invalid state %s to be receiving a PONG frame", state));
-        }
-    }
-
-    public void processText(final WsURLConnectionImpl connection,
-                            final Data dataFrame,
-                            final WebSocketFrameConsumer terminalConsumer) throws IOException {
-        WebSocketExtensionSpi sentinel = new WebSocketExtensionSpi() {
-            {
-                onTextFrameReceived = terminalConsumer;
+            switch (opcode) {
+            case BINARY:
+                transition(connection, RECEIVE_BINARY_FRAME);
+                context.onBinaryReceived(frame);
+                break;
+            case CLOSE:
+                transition(connection, RECEIVE_CLOSE_FRAME);
+                context.onCloseReceived(frame);
+                break;
+            case CONTINUATION:
+                transition(connection, RECEIVE_CONTINUATION_FRAME);
+                context.onContinuationReceived(frame);
+                break;
+            case PING:
+                transition(connection, RECEIVE_PING_FRAME);
+                context.onPingReceived(frame);
+                break;
+            case PONG:
+                transition(connection, RECEIVE_PONG_FRAME);
+                context.onPongReceived(frame);
+                break;
+            case TEXT:
+                transition(connection, RECEIVE_TEXT_FRAME);
+                context.onTextReceived(frame);
+                break;
             }
-        };
-        WebSocketContext context = connection.getContext(sentinel, false);
-        WebSocketState state = connection.getInputState();
-
-        switch (state) {
-        case OPEN:
-            transition(connection, WebSocketTransition.RECEIVED_TEXT_FRAME);
-            context.onTextReceived(dataFrame);
             break;
         default:
-            transition(connection, WebSocketTransition.ERROR);
-            context.onError(format("Invalid state %s to be receiving a TEXT frame", state));
+            transition(connection, ERROR);
+            context.onError(format("Invalid state %s to be receiving a BINARY frame", state));
```

Comment:
Missing explicit `break;` from the default case, just defensive for scenario where cases are reordered and missing break causes unexpected fall-through to next case.